pkg/connector/login.go

```diff
@@ -137,6 +141,70 @@ func (m *MetaCookieLogin) SubmitCookies(ctx context.Context, strCookies map[stri
 
 	FBID := m.getFBID(currentUser, initialTable)
 
+	// Store the cookies in the database
+
+	q := m.Main.store.GetSessionQuery()
+	session, err := q.GetByMetaID(ctx, FBID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get existing session: %w", err)
+	}
+	if session != nil {
+		log.Debug().Int64("fbid", FBID).Msg("Deleting existing session")
+		err := q.Delete(ctx, FBID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to delete existing session: %w", err)
+		}
+	}
+
+	newSession := store.MetaSession{
+		MetaID:  FBID,
+		Cookies: c,
+	}
+	err = q.Insert(ctx, &newSession)
+	if err != nil {
+		return nil, fmt.Errorf("failed to save new session: %w", err)
+	}
+
+	// Store the association between the Matrix user and the cookies
+
+	login_id := networkid.UserLoginID(fmt.Sprint(FBID))
+
+	ul, err := m.Main.Bridge.GetExistingUserLoginByID(ctx, login_id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get existing login: %w", err)
+	}
+	if ul != nil && ul.UserMXID != m.User.MXID {
+		// TODO: Do we actually want to do this?
+		ul.Delete(ctx, status.BridgeState{StateEvent: status.StateLoggedOut, Error: "overridden-by-another-user"}, false)
+		ul = nil
+	}
+	if ul == nil {
+		ul, err = m.User.NewLogin(ctx, &database.UserLogin{
+			ID: login_id,
+			Metadata: database.UserLoginMetadata{
+				StandardUserLoginMetadata: database.StandardUserLoginMetadata{
+					// TODO: Is this the correct metadata? Any extra?
+					RemoteName: currentUser.GetName(),
```

Comment:
do we need to store this? What is the point of this login metadata, where is it used?